### PR TITLE
Remove Dev UI and other Dev Mode classes from prod

### PR DESCRIPTION
**Pull request by phillip-kruger**
This PR add removing all the non-relevant dev mode classes from Prod (Dev UI Json RPC Services and HotReloadSetup etc.)

The way that you add a JSON-RPC Service in Dev UI:

```
@BuildStep(onlyIf = IsDevelopment.class)
JsonRPCProvidersBuildItem createJsonRPCServiceForCache() {
    return new JsonRPCProvidersBuildItem(CacheJsonRPCService.class);
}
```

You will do a similar thing in Prod mode only:

```
@BuildStep(onlyIf = IsNormal.class)
DevModeCleanupBuildItem cleanProd() {
    return new DevModeCleanupBuildItem(CacheJsonRPCService.class, true);
}
```

This will remove the JsonRPC service when Prod mode is build, and (due to the true) also any other classes in that package.

### CODE REVIEW

**Review by maxandersen — COMMENTED** [review]
This for me seem like a natural progression :)

I never liked that devui 2 added content to product jars that is not needed in production.

Since this is using existing mechanics I think its a good move.

I would like @aloubyansky or @dmlloyd to review it.

File: docs/src/main/asciidoc/dev-ui.adoc [hunk 1]
```diff
@@ -887,6 +887,19 @@ JsonRPCProvidersBuildItem createJsonRPCServiceForCache() {// <2>
 
 https://github.com/quarkusio/quarkus/blob/main/extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/devui/CacheDevUiProcessor.java[Example code]
 
+Also cleanup this in Prod mode, so when the app build in prod mode, we remove all dev-ui related runtime classes:
```
Comment (maxandersen) [inline docs/src/main/asciidoc/dev-ui.adoc:890]: this pops up kinda in the middle in jsonrpc explanation. Maybe have a more explicit section/title like "Remove devUI from Production" ?

File: docs/src/main/asciidoc/dev-ui.adoc [hunk 2]
```diff
@@ -887,6 +887,19 @@ JsonRPCProvidersBuildItem createJsonRPCServiceForCache() {// <2>
...
+[source,java]
+----
+@BuildStep(onlyIf = IsNormal.class)// <1>
+DevModeCleanupBuildItem cleanProd() {// <2>
+    return new DevModeCleanupBuildItem(CacheJsonRPCService.class, true);// <3>
+}
+----
+<1> Always only do this in Normal (Prod) Mode
+<2> Produce or return a `DevModeCleanupBuildItem`
+<3> Define the class or classes in your runtime module that will be removed. You can also clean the whole package of that class in case you used more than just the JSON-RPC Service
```
Comment (maxandersen) [inline docs/src/main/asciidoc/dev-ui.adoc:901]: show how to clean the whole package as otherwise one is left guessing :)

**Comment by aloubyansky** [comment]
Oh, this doesn't look like a proper way of fixing the issue. Ideally, these classes/dependencies should be properly isolated. Was there a discussion about this?

File: docs/src/main/asciidoc/dev-ui.adoc (hunk 1 above)
Comment (phillip-kruger) [reply docs/src/main/asciidoc/dev-ui.adoc:890]: It only really apply to Dev UI that has jsonrpc services. Many extensions does not have any. Any screen (js) and any build time data or runtime against the deployment cp is already not included in the prod build, it's just when you do the jsonrpc service, hence its added here

**Comment by phillip-kruger** [comment]
> Oh, this doesn't look like a proper way of fixing the issue. Ideally, these classes/dependencies should be properly isolated. Was there a discussion about this?

There was no discussion, there was an existing way to remove classes when packaging, so I just used that.

**Pull request closed without merging**